app/controllers/pensions/annualAllowance/PstrSummaryController.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.pensions.annualAllowance
+
+import config.AppConfig
+import controllers.pensions.routes.PensionsSummaryController
+import controllers.predicates.AuthorisedAction
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.PensionSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.pensions.annualAllowance.PstrSummaryView
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class PstrSummaryController @Inject()(implicit val cc: MessagesControllerComponents,
+                                      authAction: AuthorisedAction,
+                                      pstrSummaryView: PstrSummaryView,
+                                      appConfig: AppConfig,
+                                      pensionSessionService: PensionSessionService) extends FrontendController(cc) with I18nSupport {
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
+      case Some(data) =>
+        data.pensions.pensionsAnnualAllowances.pensionSchemeTaxReference match {
+          case Some(list) => Future(Ok(pstrSummaryView(taxYear, list)))
```

Comment:
I am not sure this is correct.
It's valid to not have a PSTR list on the API (it's an optional list) but still have the amounts.
This means it's possible to link to this page from CYA page with no PSTR's and a change link to add them.
At the moment it would just redirect you back to the CYA page and you can't add any PSTRs and you are stuck

This begs the question if the content should be different if no PSTRs (feels like you wouldn't render the list if count was zero  or List.Empty but just show the add button.

Reply:
Changed so it takes the pstrList, or an empty Seq if None